internal/engine/wazevo/ssa/instructions.go

```diff
@@ -851,6 +869,15 @@ var instructionSideEffects = [opcodeEnd]sideEffect{
 	OpcodeFabs:               sideEffectNone,
 	OpcodeFcopysign:          sideEffectNone,
 	OpcodeVconst:             sideEffectNone,
+	OpcodeVbor:               sideEffectNone,
+	OpcodeVbxor:              sideEffectNone,
+	OpcodeVband:              sideEffectNone,
+	OpcodeVbandnot:           sideEffectNone,
+	OpcodeVbnot:              sideEffectNone,
+	OpcodeVbitselect:         sideEffectStrict,
```

Comment:
this took a me little to figure out :^)

Reply:
why do you need strict? how is this order dependent?

Reply:
ha, good point. I picked "strict" only because it does not really trap, it's only effectful. I did not realize that was the most general one.

Reply:
fixed

Reply:
ok for reference: it wasn't really fixed, and it was not due to the op being effectful. Well, not in the meaning I intended.

BSL overwrites one of its arguments (register `rd` which both contains the input mask _and_ is used as the output register); however, ssa being an ssa _always_ want an assignment (well, duh); however, register v5 was being killed because now:

```
[[[after lowering for [4/5] "bitselect" ]]]
L1 (SSA Block: blk0):
	mov v2?.16b, v0.16b
	mov v3?.16b, v1.16b
	mov v4?.16b, v2.16b
	bsl v4?.16b, v2?.16b, v3?.16b
	mov v0.16b, v5?.16b
	ret
```

with v5 being never defined.

so the fix is to add a  `mov v5, v4` so that now v5 is defined once (then the optimizer clears out all redundant moves)